### PR TITLE
Enable cargo-spellcheck for node/forest_libp2p 

### LINKED ISSUE #1811: Enable `cargo-spellcheck` for `node/forest_libp2p`
**Issue summary**

Make sure that `cargo spellcheck --recursive node/forest_libp2p` can run without errors. For details, see the tracking issue.

Also update the spellcheck step in `.github/workflows/rust.yml` to include `node/forest_libp2p` in the set of checked folders.

**Other information and links**

### PR DESCRIPTION
**Pull request by tyshko5**
**Summary of changes**
Changes introduced in this pull request:
- 

**Reference issue to close (if applicable)**

Closes #1811

**Other information and links**

### CODE REVIEW

File: node/forest_libp2p/src/behaviour.rs [hunk 1]
```diff
@@ -490,7 +490,7 @@ impl ForestBehaviour {
         }
     }
 
-    /// Bootstrap Kademlia network
+    /// Bootstrap `Kademlia` network
```
Comment (lemmih) [inline node/forest_libp2p/src/behaviour.rs:493]: We have Kademlia in the dictionary.

File: node/forest_libp2p/src/behaviour.rs [hunk 2]
```diff
@@ -100,7 +100,7 @@ struct RequestProcessingOutcome {
     response: ChainExchangeResponse,
 }
 
-/// Event type which is emitted from the [`ForestBehaviour`] into the libp2p service.
+/// Event type which is emitted from the [`ForestBehaviour`] into the Libp2p service.
```
Comment (lemmih) [inline node/forest_libp2p/src/behaviour.rs:103]: I think you've used `LibP2P` in other PRs.

File: .config/forest.dic [hunk 3]
```diff
@@ -36,10 +36,10 @@ deserializing
 mempool
 sortition
 stateful
+MDNS
+Libp2p
```
Comment (lemmih) [inline .config/forest.dic:40]: There's already `LibP2P` in the dictionary.

**Review by lemmih — CHANGES_REQUESTED** [review]

File: node/forest_libp2p/src/config.rs [hunk 4]
```diff
@@ -14,7 +14,7 @@ pub struct Libp2pConfig {
     pub bootstrap_peers: Vec<Multiaddr>,
     /// MDNS discovery enabled.
     pub mdns: bool,
-    /// Kademlia discovery enabled.
+    /// `Kademlia` discovery enabled.
```
Comment (lemmih) [inline node/forest_libp2p/src/config.rs:17]: Kademlia is in the dictionary.

File: node/forest_libp2p/src/discovery.rs [hunk 5]
```diff
@@ -37,7 +37,7 @@ pub enum DiscoveryOut {
 
 /// `DiscoveryBehaviour` configuration.
 ///
-/// Note: In order to discover nodes or load and store values via Kademlia one has to add at least
+/// Note: In order to discover nodes or load and store values via `Kademlia` one has to add at least
```
Comment (lemmih) [inline node/forest_libp2p/src/discovery.rs:40]: Kademlia is in the dictionary.

File: node/forest_libp2p/src/discovery.rs [hunk 6]
```diff
@@ -82,7 +82,7 @@ impl<'a> DiscoveryConfig<'a> {
         self
     }
 
-    /// Configures if Kademlia is enabled.
+    /// Configures if `Kademlia` is enabled.
```
Comment (lemmih) [inline node/forest_libp2p/src/discovery.rs:85]: Kademlia is in the dictionary.

File: node/forest_libp2p/src/discovery.rs [hunk 7]
```diff
@@ -167,11 +167,11 @@ pub struct DiscoveryBehaviour {
     /// User-defined list of nodes and their addresses. Typically includes bootstrap nodes and
     /// reserved nodes.
     user_defined: Vec<(PeerId, Multiaddr)>,
-    /// Kademlia discovery.
+    /// `Kademlia` discovery.
```
Comment (lemmih) [inline node/forest_libp2p/src/discovery.rs:170]: Kademlia is in the dictionary.

File: node/forest_libp2p/src/discovery.rs [hunk 8]
```diff
@@ -167,11 +167,11 @@ pub struct DiscoveryBehaviour {
...
     kademlia: Toggle<Kademlia<MemoryStore>>,
     /// Discovers nodes on the local network.
     mdns: Toggle<Mdns>,
-    /// Stream that fires when we need to perform the next random Kademlia query.
+    /// Stream that fires when we need to perform the next random `Kademlia` query.
```
Comment (lemmih) [inline node/forest_libp2p/src/discovery.rs:174]: Kademlia is in the dictionary.

**Review by lemmih — CHANGES_REQUESTED** [review]

**Review by lemmih — APPROVED** [review]

**Review by jdjaustin — APPROVED** [review]

**Pull request merged**